include/pybind11/detail/internals.h

```diff
@@ -314,16 +314,44 @@ inline internals **&get_internals_pp() {
     return internals_pp;
 }
 
-// forward decl
-inline void translate_exception(std::exception_ptr);
+std::forward_list<ExceptionTranslator> &get_exception_translators();
+std::forward_list<ExceptionTranslator> &get_local_exception_translators();
+
+// Apply all the extensions translators from a list
+// Return true if one of the translators completed without raising an exception
+// itself. Return of false indicates that if there are other translators
+// available, they should be tried.
+inline bool apply_exception_translators(std::forward_list<ExceptionTranslator> &translators,
```

Comment:
The args to these can actually be const ref, can't they?

Reply:
Right, that was an oversight on my end. Just pushed another revision.